lib/spack/docs/packaging_guide.rst

```diff
@@ -2054,19 +2147,15 @@ The parameters are as follows:
     targets into.  It acts like a string, but it's actually its own
     special type, :py:class:`Prefix <spack.util.prefix.Prefix>`.
 
-``spec`` and ``prefix`` are passed to ``install`` for convenience.
-``spec`` is also available as an attribute on the package
-(``self.spec``), and ``prefix`` is actually an attribute of ``spec``
-(``spec.prefix``).
+The arguments ``spec`` and ``prefix`` are passed only for convenience, as they always
+correspond to ``self.spec`` and ``self.spec.prefix`` respectively.
```

Comment:
Can this doc just explain what `spec` and `prefix` are?  "`spec` is the same as `self.spec`" is useful to know (and should not be removed), but by itself is somewhat circular.